etcdserver/api/v2http/client.go

```diff
@@ -761,6 +761,9 @@ func trimErrorPrefix(err error, prefix string) error {
 
 func unmarshalRequest(r *http.Request, req json.Unmarshaler, w http.ResponseWriter) bool {
 	ctype := r.Header.Get("Content-Type")
+	if strings.Contains(ctype, ";") {
```

Comment:
go stdlib has a more complicated version. We probably do not need that full version, but

``` go
i := strings.Index(ctype, ";")
if i != -1 {
   ctype = strings.TrimSpace(strings.ToLower(ctype[0:i]))
}
```
seems to be better. It is case insensitive.

see https://golang.org/src/mime/mediatype.go?s=2584:2669#L94